midterm-2023-11-07/code/5311-kristijan-lazarov/script.ts

```diff
@@ -0,0 +1,322 @@
+    }
+
+    displayMovies(filteredMovies);
+}
+
+const applyNameFilter = () => {
+    const nameFilter = document.getElementById('name-filter') as HTMLInputElement;
+    const searchQuery = nameFilter.value.toLowerCase();
+
+    let filteredMovies = movies;
+
+    if (searchQuery.trim() !== '') {
+        filteredMovies = filteredMovies.filter(movie => movie.title.toLowerCase().includes(searchQuery));
+    }
+
+    displayMovies(filteredMovies);
+}
+
+const applyActorFilter = () => {
+    const actorFilter = document.getElementById('actor-filter') as HTMLInputElement;
+    const searchQuery = actorFilter.value.toLowerCase();
+
+    let filteredMovies = movies;
+
+    if (searchQuery.trim() !== '') {
+        filteredMovies = filteredMovies.filter(movie => movie.cast.some(actor => actor.actor.toLowerCase().includes(searchQuery)));
+    }
+
+    displayMovies(filteredMovies);
+}
+
+const loadData = async () => {
+    const dataUri = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-07/data/movies.json";
+    const response = await fetch(dataUri);
+
+    if (!response.ok) {
+        throw new Error("The data is not there");
+    }
+
+    const data = await response.json();
+    return data;
+}
+
+
+function displayMovies(moives: Movie[]) {
+
+    const container = document.getElementById("movie-container")!;
+    container.innerHTML = "";
+    for (const movie of moives) {
+        const movieRow = generateMovieRow(movie);
+        container.appendChild(movieRow);
+    }
+
+}
+
+const generateMovieRow = (movie: Movie) => {
+    const row = document.createElement("div");
+    row.classList.add("movie-row");
+
+    //id
+    const idCell = document.createElement("div");
+    idCell.classList.add("movie-data", "movie-id");
+    idCell.innerHTML = movie.id.toString();
+    row.appendChild(idCell);
+
+
+    //name
+    const nameCell = document.createElement("div");
+    nameCell.classList.add("movie-data", "movie-name");
+    nameCell.innerHTML = movie.title;
+    row.appendChild(nameCell);
+
+
+    //director
+    const directorCell = document.createElement("div");
+    directorCell.classList.add("movie-data", "movie-director");
+    directorCell.innerHTML = movie.director;
+    row.appendChild(directorCell);
+
+
+    //year
+    const relaseCell = document.createElement("div");
+    relaseCell.classList.add("movie-data", "movie-relese");
+    relaseCell.innerHTML = movie.year.toString();
+    row.appendChild(relaseCell);
+
+
+    //genre
+    const genreCell = document.createElement("div");
+    genreCell.classList.add("movie-data", "movie-genre");
+    genreCell.innerHTML = movie.genre.join(" / ");
+    row.appendChild(genreCell);
+
+
+    //plot
+    const plotCell = document.createElement("div");
+    plotCell.classList.add("movie-data", "movie-plot");
+    const maxCharacters = 50;
+    let content = movie.plot;
+
+    if (content.length > maxCharacters) {
```

Comment:
The actual data extraction should be moved to a function, so that it does not mess up the row generation.